plextraktsync/trakt_api.py

```diff
@@ -546,11 +546,24 @@ def from_guid(self, guid: PlexGuid):
         Find Trakt Episode from Guid of Plex Episode
         """
         te = self.from_number(guid.pm.season_number, guid.pm.episode_number)
-        if not te or (str(te.ids.get(guid.provider)) != guid.id and not guid.pm.is_legacy_agent):
+        if self.invalid_match(guid, te):
             te = self.from_id(guid.provider, guid.id)
 
         return te
 
+    @staticmethod
+    def invalid_match(guid: PlexGuid, episode: Optional[TVEpisode]) -> bool:
+        """
+        Checks if guid and episode don't match by comparing trakt provided id
+        """
+
+        if not episode:
+            return True
+        reverse_id = getattr(episode, guid.provider, None)
+        if reverse_id != guid.id and not guid.pm.is_legacy_agent:
```

Comment:
btw, why the `guid.pm.is_legacy_agent` check is relevant?

Reply:
I think it's because of old tv guids like `987654/1/1` which is not a real episode id

Reply:
so, the check could be simplified to this?

```diff
--- a/plextraktsync/trakt_api.py
+++ b/plextraktsync/trakt_api.py
@@ -557,10 +557,10 @@ def invalid_match(guid: PlexGuid, episode: Optional[TVEpisode]) -> bool:
         Checks if guid and episode don't match by comparing trakt provided id
         """

-        if not episode:
+        if not episode or guid.pm.is_legacy_agent:
             return True
         reverse_id = getattr(episode, guid.provider, None)
-        if str(reverse_id) != guid.id and not guid.pm.is_legacy_agent:
+        if str(reverse_id) != guid.id:
             return True
         return False
```

or rather (I'm bad ad logic inversion expressions)?

```diff
--- a/plextraktsync/trakt_api.py
+++ b/plextraktsync/trakt_api.py
@@ -559,8 +559,10 @@ def invalid_match(guid: PlexGuid, episode: Optional[TVEpisode]) -> bool:

         if not episode:
             return True
+        if guid.pm.is_legacy_agent:
+            return False
         reverse_id = getattr(episode, guid.provider, None)
-        if str(reverse_id) != guid.id and not guid.pm.is_legacy_agent:
+        if str(reverse_id) != guid.id:
             return True
         return False

```

Reply:
And `reverse_id` is not a good name, it should be named `id_from_trakt` or `trakt_episode_id` because we compare the id from trakt API with plex guid.id

Reply:
256dec0b, 06eb4093